raiden/network/transport/matrix/transport.py

```diff
@@ -1053,13 +1060,19 @@ def _get_private_room(self, invitees: List[User]) -> Room:
         self.log.debug("Creating private room", room=room, invitees=invitees)
         return room
 
-    def _get_public_room(self, room_name: str, invitees: List[User]) -> Room:
+    def _get_public_room(self, invitees: List[User]) -> Room:
         """ Obtain a public, canonically named (if possible) room and invite peers """
+        # retrieve address from invitees implicitly, is constant for all invitees
+        assert self._raiden_service is not None  # Extra assertion for mypy
+        address = validate_userid_signature(invitees[0])
```

Comment:
This deserves an assert. Please make sure all users have the same address after validated.

Just a note: I'm assuming an attacker cannot put an invalid `User` into this list by joining a public room.

Reply:
> I believe this condition here is incorrect. It is perfectly fine for a node to go offline, and if that is the case I don't think a new room should be created.

What I wanted to do here is go in the elif after we checked all rooms for online users and return one without an online user. There should not be a room created in case there is one for the peer already- Instead one without an online user would be returned  by the elif a few lines later.

Reply:
>  I'm assuming an attacker cannot put an invalid User into this list by joining a public room.
This is possible, by squatting the name or by replaying the displayname, in both cases validate would return None so I'd rather not assert here